### PR TITLE
CIRC-1143 Implement two overriding blocks for renewal overriding with/without DueDate field

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Approach
- when trying to renew a loan, all blocks should be validated and returned an error with block depending on a new due date is required or not;
- add renewalDueDateRequiredBlock, update schemas, update logic in RenewalResource;
- update tests;

Resolves: [CIRC-1143](https://issues.folio.org/browse/CIRC-1143)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [x] Were there any schema changes?
  - [x] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [x] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -150,7 +150,7 @@
     },
     {
       "id": "circulation",
-      "version": "10.0",
+      "version": "11.1",
```
Comment (marcjohnson-kint) [inline descriptors/ModuleDescriptor-template.json:153]: Why has this jumped a major and minor version in one step?

Comment (roman-barannyk) [reply descriptors/ModuleDescriptor-template.json:153]: This PR was based on branch CIRC-1091, which has not been merged yet. In CIRC-1091 the version was increased to 11.0, and this PR increases it to 11.1. I'm going to merge them sequentially: CIRC-1091 and then CIRC-1143

Comment (marcjohnson-kint) [reply descriptors/ModuleDescriptor-template.json:153]: > This PR was based on branch CIRC-1091, which has not been merged yet. In CIRC-1091 the version was increased to 11.0, and this PR increases it to 11.1. I'm going to merge them sequentially: CIRC-1091 and then CIRC-1143

Ok, thank you for explaining that

File: src/main/java/org/folio/circulation/resources/renewal/RenewalResource.java [hunk 2]
```diff
@@ -135,6 +184,34 @@ private void renew(RoutingContext routingContext) {
       .thenAccept(webContext::writeResultToHttpResponse);
   }
 
+  private CompletableFuture<Result<RenewalContext>> processFeesFines(RenewalContext renewalContext,
+    Clients clients) {
+
+    return isRenewalBlockOverrideRequested
+      ? processFeesFinesForOverriding(renewalContext, clients)
+      : processFeesFinesForRegularRenew(renewalContext, clients);
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/renewal/RenewalResource.java:192]: processFeesFinesForOverriding -> processFeesFinesForRenewalBlockOverride
processFeesFinesForRegularRenew -> processFeesFines

File: src/main/java/org/folio/circulation/resources/renewal/RenewalResource.java [hunk 3]
```diff
@@ -222,17 +301,300 @@ private HttpResponse toResponse(JsonObject body) {
...
+      renewDate);
+    final List<ValidationError> errors = new ArrayList<>();
+    addErrorsIfDueDateResultFailed(loan, errors, proposedDueDateResult);
+
+    if (errors.isEmpty()) {
+      final BlockOverrides blockOverrides = BlockOverrides.from(getObjectProperty(
+        context.getRenewalRequest(), "overrideBlocks"));
+
+      if (!blockOverrides.getPatronBlockOverride().isRequested() &&
+        !blockOverrides.getRenewalBlockOverride().isRequested()) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/renewal/RenewalResource.java:494]: Can we use class field for this?

File: src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java [hunk 4]
```diff
@@ -320,7 +320,7 @@ private LoanPolicyBuilder rollingPolicy() {
   }
 
   private Result<Loan> renew(RenewalContext context) {
-    return new OverrideRenewalStrategy().renew(context, null)
+    return new RenewByBarcodeResource(null).renewThroughOverride(context)
```
Comment (alexanderkurash) [inline src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java:323]: Merge this class and RegularRenewalStrategyTest.java.

File: src/test/java/org/folio/circulation/resources/renewal/RegularRenewalStrategyTest.java [hunk 5]
```diff
@@ -66,46 +65,58 @@ public void cannotRenewWhenRecallRequestedAndPolicyNorLoanableAndItemLost() {
...
-      hasMessage("item is Aged to lost")
-    ));
+    CirculationErrorHandler errorHandler = new OverridingErrorHandler(null);
+    renew(loan, recallRequest, errorHandler);
+
+    assertEquals(3, errorHandler.getErrors().size());
+    assertTrue(errorHandler.getErrors().keySet().stream()
+      .map(ValidationErrorFailure.class::cast)
+      .anyMatch(httpFailure -> httpFailure.hasErrorWithReason(
+        "items cannot be renewed when there is an active recall request")));
```
Comment (alexanderkurash) [inline src/test/java/org/folio/circulation/resources/renewal/RegularRenewalStrategyTest.java:75]: Extract method

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**